### PR TITLE
Dave's Concrete

### PR DESCRIPTION
**Pull request by BadIsMyUsername**
## About The Pull Request
_NOTE TO FELLOW CODERS - Values are rough and shaky right now and I will need help adjusting them before merge_

- Adds the ability to create concrete walls such as the supermart and store wall, as well as a new concrete barricade. 

- Adds limestone chunks to dig loot, which can be ground up into calcium carbonate, a reagent needed to create cement (which then makes concrete). Sand can now also be ground up into silicon, rather than just glass.

- Adds the calcium carbonate reagent, cement reagent, and concrete chemical receipes. 

- Adds a new drink, the Concrete Shake, to celebrate the rediscovery of concrete!

- Raises prob for desert dig loot to appear slightly (from prob(1) to prob(2))

## Why It's Good For The Game
It has always been strange to me that we can create many things such as advanced technological weapons (plasma rifles) yet have lost the ability to make simple concrete structures. Now, patching up the Followers walls with ugly metal instead of somewhat less ugly supermart walls is no longer needed! 
Dave's Concrete
The steps required to produce concrete can be followed by anyone without a chem machine (just a welder, spade, and an all-in-one grinder).

**_Dusty Dave's Guide to Concrete Construction_**
Calcium carbonate = Milk 1u + Carbon 1u + Oxygen 3u, heated to 375k
Cement = Calcium carbonate 6u + Silicon 3u + Iron 1u, heated to 750k
Concrete = Cement 1u + Water 2u + Silicon 7u. _**Makes one "bag" (or sheet) of concrete.**_

Make barricades with concrete in hand. Coat barricades with more concrete to make store walls.
Coat girders with concrete to make supermart walls.

## Changelog
:cl:
add: Added a lot of concrete things
/:cl:

### CODE REVIEW

**Comment by Pebblesthe2nd** [comment]
Checking for space indentation
code/game/turfs/f13concrete.dm: //Makes one concrete sheet
space indentation detected

Did this compile locally? Usually the compiler catches space indentation errors and give the line number.
If not already using Visual Studio Code I heartily recommend it.

**Comment by BadIsMyUsername** [comment]
> Did this compile locally? Usually the compiler catches space indentation errors and give the line number.
> If not already using Visual Studio Code I heartily recommend it.

Right now I've been using the BUILD.bat compiler provided, but have used VS in the past. Will probably switch to using that soon instead of just pure DM haha.

File: code/game/turfs/f13concrete.dm [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+	name = "supermart wall"
+	desc = "A pre-War supermart wall made of reinforced concrete. This one looks newly built"
+	icon = 'icons/turf/walls/f13superstore.dmi'
+	icon_state = "supermart"
+	icon_type_smooth = "supermart"
+	hardness = 90
+	explosion_block = 2
+	smooth = SMOOTH_TRUE
+	sheet_type = /obj/item/stack/sheet/mineral/concrete
+	canSmoothWith = list(/turf/closed/wall/f13/supermart, /turf/closed/wall/mineral/concrete, /turf/closed/wall)
```
Comment (Pebblesthe2nd) [inline code/game/turfs/f13concrete.dm:170]:
```suggestion
	canSmoothWith = list(/turf/closed/wall/f13/supermart, /turf/closed/wall/mineral/concrete, /turf/closed/wall,)
```

File: code/game/turfs/f13concrete.dm [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+////////////
+/datum/reagent/concrete
+	name = "Concrete"
+	description = "Concrete"
+	color = "#A49483"
+
+/datum/chemical_reaction/concrete
+	name = "concrete"
+	id = "concrete"
+	required_reagents = list(/datum/reagent/cement = 1, /datum/reagent/water = 2, /datum/reagent/silicon = 7)
```
Comment (Pebblesthe2nd) [inline code/game/turfs/f13concrete.dm:112]:
```suggestion
	required_reagents = list(/datum/reagent/cement = 1, /datum/reagent/water = 2, /datum/reagent/silicon = 7,)
```

File: code/game/turfs/f13concrete.dm [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+*/
+
+/////////////////////////////////
+//Limestone (calcium carbonate)//
+/////////////////////////////////
+/datum/chemical_reaction/calciumcarbonate
+	name = "calcium carbonate"
+	id = "calciumcarbonate"
+	results = list(/datum/reagent/calciumcarbonate = 5)
+	required_reagents = list(/datum/reagent/consumable/milk = 1, /datum/reagent/carbon = 1, /datum/reagent/oxygen = 3)
```
Comment (Pebblesthe2nd) [inline code/game/turfs/f13concrete.dm:21]:
```suggestion
	required_reagents = list(/datum/reagent/consumable/milk = 1, /datum/reagent/carbon = 1, /datum/reagent/oxygen = 3,)
```

File: code/game/turfs/f13concrete.dm [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+
+
+//////////
+//Cement//
+//////////
+/datum/chemical_reaction/cement
+	name = "cement"
+	id = "cement"
+	results = list(/datum/reagent/cement = 10)
+	required_reagents = list(/datum/reagent/calciumcarbonate = 6, /datum/reagent/silicon = 3, /datum/reagent/iron = 1)
```
Comment (Pebblesthe2nd) [inline code/game/turfs/f13concrete.dm:58]:
```suggestion
	required_reagents = list(/datum/reagent/calciumcarbonate = 6, /datum/reagent/silicon = 3, /datum/reagent/iron = 1,)
```

File: code/game/turfs/f13concrete.dm [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+	..()
+
+//////////////////
+//Concrete Shake//
+//////////////////
+/datum/chemical_reaction/concreteshake
+	name = "Concrete Shake"
+	id = "concrete_shake"
+	results = list(/datum/reagent/consumable/concreteshake = 5)
+	required_reagents = list(/datum/reagent/consumable/milk = 1, /datum/reagent/consumable/sugar = 1, /datum/reagent/cement = 1, /datum/reagent/consumable/ice = 1, /datum/reagent/consumable/cream = 1)
```
Comment (Pebblesthe2nd) [inline code/game/turfs/f13concrete.dm:81]:
```suggestion
	required_reagents = list(/datum/reagent/consumable/milk = 1, /datum/reagent/consumable/sugar = 1, /datum/reagent/cement = 1, /datum/reagent/consumable/ice = 1, /datum/reagent/consumable/cream = 1,)
```

File: code/game/turfs/f13concrete.dm [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+
+/obj/item/stack/sheet/mineral/concrete/five
+	amount = 5
+
+/obj/item/stack/sheet/mineral/concrete/get_main_recipes()
+	. = ..()
+	. += GLOB.concrete_recipes
+
+GLOBAL_LIST_INIT(concrete_recipes, list ( \
+	new/datum/stack_recipe("concrete barricade", /obj/structure/barricade/concrete, 2, time = 30, one_per_turf = 1, on_floor = 1), \
```
Comment (Pebblesthe2nd) [inline code/game/turfs/f13concrete.dm:149]: Not sure what the 2 in this line does but I will assume its something intended

File: code/game/turfs/f13concrete.dm [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+	name = "store wall"
+	desc = "A pre-War store wall made of solid concrete. This one looks newly built."
+	icon = 'icons/turf/walls/f13store.dmi'
+	icon_state = "store"
+	icon_type_smooth = "store"
+	hardness = 80
+	smooth = SMOOTH_TRUE
+	girder_type = /obj/structure/barricade/concrete
+	sheet_type = null
+	canSmoothWith = list(/turf/closed/wall/f13/store, /turf/closed/wall/f13/store/constructed, /turf/closed/wall)
```
Comment (Pebblesthe2nd) [inline code/game/turfs/f13concrete.dm:219]:
```suggestion
	canSmoothWith = list(/turf/closed/wall/f13/store, /turf/closed/wall/f13/store/constructed, /turf/closed/wall,)
```

File: code/modules/fallout/turf/walls.dm [hunk 8]
```diff
@@ -115,7 +115,7 @@ turf/closed/wall/f13/wood/house/update_damage_overlay()
 	baseturfs = /turf/open/indestructible/ground/outside/ruins
 	girder_type = 0
 	sheet_type = null
-	canSmoothWith = list(/turf/closed/wall/f13/store, /turf/closed/wall)
+	canSmoothWith = list(/turf/closed/wall/f13/store, /turf/closed/wall/f13/store/constructed, /turf/closed/wall)
```
Comment (Pebblesthe2nd) [inline code/modules/fallout/turf/walls.dm:118]:
```suggestion
	canSmoothWith = list(/turf/closed/wall/f13/store, /turf/closed/wall/f13/store/constructed, /turf/closed/wall,)
```

File: code/modules/fallout/turf/walls.dm [hunk 9]
```diff
@@ -144,7 +144,7 @@ turf/closed/wall/f13/wood/house/update_damage_overlay()
 	//	disasemblable = 0
 	girder_type = 0
 	sheet_type = null
-	canSmoothWith = list(/turf/closed/wall/f13/supermart, /turf/closed/wall)
+	canSmoothWith = list(/turf/closed/wall/f13/supermart, /turf/closed/wall/mineral/concrete, /turf/closed/wall)
```
Comment (Pebblesthe2nd) [inline code/modules/fallout/turf/walls.dm:147]:
```suggestion
	canSmoothWith = list(/turf/closed/wall/f13/supermart, /turf/closed/wall/mineral/concrete, /turf/closed/wall,)
```

File: code/modules/mining/ores_coins.dm [hunk 10]
```diff
@@ -102,6 +102,7 @@
 	refined_type = /obj/item/stack/sheet/glass
 	w_class = WEIGHT_CLASS_TINY
 	merge_type = /obj/item/stack/ore/glass
+	grind_results = list(/datum/reagent/silicon = 20)
```
Comment (Pebblesthe2nd) [inline code/modules/mining/ores_coins.dm:105]:
```suggestion
	grind_results = list(/datum/reagent/silicon = 20,)
```

File: code/modules/fallout/turf/ground.dm [hunk 11]
```diff
@@ -147,7 +148,7 @@
 
 /turf/open/indestructible/ground/outside/desert/Initialize()
 	. = ..()
-	if(prob(1))
+	if(prob(2))
```
Comment (Pebblesthe2nd) [inline code/modules/fallout/turf/ground.dm:151]: This doubles the chance of finding stuff with a metal detector correct?

**Review by Pebblesthe2nd — CHANGES_REQUESTED** [review]
PR looks good, suggestions all about trailing commas to be up to code guidelines.
Clarification requested on the ,2, value in the global list, and the prob(2) for digging as seen in the comments, before stamping.

**Comment by Pebblesthe2nd** [comment]
> > Did this compile locally? Usually the compiler catches space indentation errors and give the line number.
> > If not already using Visual Studio Code I heartily recommend it.
> 
> Right now I've been using the BUILD.bat compiler provided, but have used VS in the past. Will probably switch to using that soon instead of just pure DM haha.

I use the BUILD.bat myself honestly, but VSC does offer some advantages for that too. Either way, the compiler misses stuff, ain't no thing. Make sure to get the Github and BYOND extensions to VSC and link your git account to get a lot of added functionality. Whatever works for you is fine.

File: code/modules/fallout/turf/ground.dm (hunk 11 above)
Comment (BadIsMyUsername) [reply code/modules/fallout/turf/ground.dm:151]: Should not double the chance, but increase it by one percent (from one percent), if my knowledge of how the prob proc works. 

From the guide:
>prob(P)
>returns with probability P percent

**Comment by BadIsMyUsername** [comment]
> PR looks good, suggestions all about trailing commas to be up to code guidelines.
Wait, do we really need to add commas to the end of all those lists? I'll do it, just seems strange to me

> Clarification requested on the ,2, value in the global list
That is how much material is required to craft the object, so 2 concrete sheets (or bags, whatever you want to call them).

File: code/game/turfs/f13concrete.dm (hunk 6 above)
Comment (BadIsMyUsername) [reply code/game/turfs/f13concrete.dm:149]: The amount of concrete needed to craft the barricade!

**Review by Pebblesthe2nd — APPROVED** [review]

**Comment by Pebblesthe2nd** [comment]
You can just go to files changed here in git and make a batch commit of all the suggestions, just  tip =)

**Comment by BadIsMyUsername** [comment]
> You can just go to files changed here in git and make a batch commit of all the suggestions, just tip =)

I got lazy haha

**Comment by Pebblesthe2nd** [comment]
Haha that percentage comment cracked me up =D
Anyways, all good here, review thumbs up, only flag thing is the digging that might cause a modest resource flooding, but its a minor concern at this point (since a number of factors already make this a thing), might as well try out the 2% in my opinion.

(Trailing commas is the new hot thing, slap it on every list and be safe. That and keeping down horizontal drift are the two rules we use most, not quite up to TG code standards but that is the goal.)

**Pull request merged**